### PR TITLE
feat: Added support for handling lowercase addresses

### PR DESCRIPTION
**Pull request by alexrisch**
Will now get checksum addresses from provided addresses Added some helper libraries to avoid adding additional dependency on ethers

Simulator Screenshot - iPhone 15 Pro - 2024-02-01 at 14 10 37

### CODE REVIEW

File: src/utils/address.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { keccak_256 } from '@noble/hashes/sha3'
```
Comment (alexrisch) [inline src/utils/address.ts:1]: Port from viem getAddress

**Comment by cameronvoell** [comment]
this change looks good, sounds like it is addressing a specific issue with  adding new conversations.

I wonder if we should add the new getAddress in a few more locations. For example if you pass in a lowercase address to client.contacts.deny in the test linked below, I think we will fail to block the address. Perhaps we should merge this change, but keep your [issue](https://github.com/xmtp/xmtp-react-native/issues/230) open until we feel we have a comprehensive list of places where checksums need to added. WDYT?

https://github.com/xmtp/xmtp-react-native/blob/c68bcc551ab7ec1eb3de8ae6ec936fb6eaff4172/example/src/tests.ts#L676

**Review by cameronvoell — APPROVED** [review]
Approved with the recommendation we keep https://github.com/xmtp/xmtp-react-native/issues/230 open for now

**Comment by alexrisch** [comment]
> this change looks good, sounds like it is addressing a specific issue with adding new conversations.
> 
> I wonder if we should add the new getAddress in a few more locations. For example if you pass in a lowercase address to client.contacts.deny in the test linked below, I think we will fail to block the address. Perhaps we should merge this change, but keep your [issue](https://github.com/xmtp/xmtp-react-native/issues/230) open until we feel we have a comprehensive list of places where checksums need to added. WDYT?
> 
> https://github.com/xmtp/xmtp-react-native/blob/c68bcc551ab7ec1eb3de8ae6ec936fb6eaff4172/example/src/tests.ts#L676

Good catch, let me figure out LOE and determine

**Comment by alexrisch** [comment]
> Approved with the recommendation we keep #230 open for now

Updated handling for contacts methods

**Review by cameronvoell — APPROVED** [review]
Changes look good, think this PR should fully close https://github.com/xmtp/xmtp-react-native/issues/230 now, nice!

**Pull request merged**